ndk/src/media/image_reader.rs

```diff
@@ -0,0 +1,309 @@
+        })?;
+
+        let image = Image { inner };
+
+        Ok(match fence.assume_init() {
+            -1 => (image, None),
+            fence => (image, Some(fence)),
+        })
+    }
+
+    pub fn acquire_latest_image(&self) -> Result<Option<Image>> {
+        let res = construct_never_null(|res| unsafe {
+            ffi::AImageReader_acquireLatestImage(self.as_ptr(), res)
+        });
+
+        if Err(NdkMediaError::ImgreaderNoBufferAvailable) == res {
+            return Ok(None);
+        }
+
+        Ok(Some(Image { inner: res? }))
+    }
+
+    /// # Safety
+    /// If the returned file descriptor is not `None`, it must be awaited before attempting to access the Image returned.
+    /// https://developer.android.com/ndk/reference/group/media#aimagereader_acquirelatestimageasync
+    #[cfg(feature = "api-level-26")]
+    pub fn acquire_latest_image_async(&self) -> Result<(Image, Option<RawFd>)> {
+        let mut fence = MaybeUninit::uninit();
+        let inner = construct_never_null(|res| unsafe {
+            ffi::AImageReader_acquireLatestImageAsync(self.as_ptr(), res, fence.as_mut_ptr())
+        })?;
+
+        let image = Image { inner };
+
+        Ok(match unsafe { fence.assume_init() } {
+            -1 => (image, None),
+            fence => (image, Some(fence)),
+        })
+    }
+}
+
+impl Drop for ImageReader {
+    fn drop(&mut self) {
+        unsafe { ffi::AImageReader_delete(self.as_ptr()) };
+    }
+}
+
+#[derive(Debug)]
+pub struct Image {
+    inner: NonNull<ffi::AImage>,
+}
+
+pub type CropRect = ffi::AImageCropRect;
+
+impl Image {
+    fn as_ptr(&self) -> *mut ffi::AImage {
+        self.inner.as_ptr()
+    }
+
+    pub fn get_plane_data(&self, plane_idx: i32) -> Result<&[u8]> {
+        let mut result_ptr = MaybeUninit::uninit();
+        let mut result_len = MaybeUninit::uninit();
+        let status = unsafe {
+            ffi::AImage_getPlaneData(
+                self.as_ptr(),
+                plane_idx,
+                result_ptr.as_mut_ptr(),
+                result_len.as_mut_ptr(),
+            )
+        };
+
+        NdkMediaError::from_status(status, || unsafe {
+            std::slice::from_raw_parts(result_ptr.assume_init(), result_len.assume_init() as _)
+        })
+    }
+
+    pub fn get_plane_pixel_stride(&self, plane_idx: i32) -> Result<i32> {
+        construct(|res| unsafe { ffi::AImage_getPlanePixelStride(self.as_ptr(), plane_idx, res) })
+    }
+
+    pub fn get_plane_row_stride(&self, plane_idx: i32) -> Result<i32> {
+        construct(|res| unsafe { ffi::AImage_getPlaneRowStride(self.as_ptr(), plane_idx, res) })
+    }
+
+    pub fn get_crop_rect(&self) -> Result<CropRect> {
+        construct(|res| unsafe { ffi::AImage_getCropRect(self.as_ptr(), res) })
+    }
+
+    pub fn get_width(&self) -> Result<i32> {
+        construct(|res| unsafe { ffi::AImage_getWidth(self.as_ptr(), res) })
+    }
+
+    pub fn get_height(&self) -> Result<i32> {
```

Comment:
it's uncommon to prefix getters with `get_`. on the other hand staying close to the original naming might also be good. I have mixed feelings on this.